src/org/gk/model/InstanceUtilities.java

```diff
@@ -1394,4 +1402,92 @@ else if (instance.getSchemClass().isa(ReactomeJavaConstants.Pathway))
 	//public static String encodeLineSeparators(String text) {
 	//    return text.replaceAll(FileAdaptor.LINE_END + "", "<br>");
 	//}
+
+     /**
+      * Determines if an instance is a drug or contains a drug (in the case of EntitySets and Complexes).
+      *
+      * @param instance
+      * @throws Exception
+      * @returns boolean
+      */
+     public static boolean isDrug(GKInstance instance) throws Exception {
+    	 if (instance == null)
+    		 return false;
+
+    	 SchemaClass schemaClass = instance.getSchemClass();
+    	 // Check if instance is a drug.
+    	 if (schemaClass.isa(ReactomeJavaConstants.Drug))
+    		 return true;
+
+    	 List<String> validClasses = Arrays.asList(ReactomeJavaConstants.EntitySet,
+												   ReactomeJavaConstants.Complex);
+    	 // Check id instance is an EntitySet or Complex.
+    	 if (validClasses.stream().noneMatch(schemaClass::isa))
+			 return false;
+    	 
+    	 // If EntitySet has a "hasMember" or "hasCandidate" attribute then recursively iterate over them.
+    	 // ifHasInstance()
+    	 Set<GKInstance> members = InstanceUtilities.getContainedInstances(instance,
+																		   ReactomeJavaConstants.hasMember,
+																		   ReactomeJavaConstants.hasCandidate,
+																		   ReactomeJavaConstants.hasComponent);
+    	 // No members.
+    	 if (members.size() == 0 || members == null)
+    		 return false;
+
+    	 // Check if instance contains a drug by:
+    	 //   (1) Iterating over all schema classes of the instance's members.
+    	 //   (2) Checking if any of the ancestor schema class is a Drug class.
+    	 return members.stream()
+					   .map(GKInstance::getSchemClass)
+					   .anyMatch(cls -> cls.isa(ReactomeJavaConstants.Drug));
+     }
+
+    private MySQLAdaptor getTestDBA() throws SQLException {
```

Comment:
Is this method called somewhere else? If not, don't add it.

I think it will be necessary for you to add another class for doing test. Otherwise, there is just too much code in the class.

Reply:
Understood — fixed in [4082ae8](https://github.com/reactome/CuratorTool/pull/23/commits/4082ae86285bba3e8a3bf57c30f85e8735ec6773).